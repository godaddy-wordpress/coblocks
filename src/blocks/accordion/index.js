/**
 * External dependencies
 */
import { AccordionIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies
 */
import edit from './edit';
import metadata from './block.json';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Icon } from '@wordpress/components';
import { InnerBlocks } from '@wordpress/block-editor';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	attributes,
	/* translators: block description */
	description: __( 'Organize content within collapsable accordion items.', 'coblocks' ),
	edit,
	example: {
		attributes: {
			count: 2,
		},
	},
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'tabs', 'coblocks' ),
		/* translators: block keyword (abbreviation for "frequently asked questions") */
		__( 'faq', 'coblocks' ),
	],
	save() {
		return (
			<div>
				<InnerBlocks.Content />
			</div>
		);
	},
	supports: {
		align: [ 'wide', 'full' ],
		html: false,
	},
	/* translators: block name */
	title: __( 'Accordion', 'coblocks' ),
	transforms,
};

export { name, category, metadata, settings };
