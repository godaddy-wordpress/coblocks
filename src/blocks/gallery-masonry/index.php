<?php
/**
 * Server-side rendering of the `core/image` block.
 *
 * @package WordPress
 */

/**
 * Handles backwards compatibility for Masonry Gallery Blocks,
 * whose images feature a `data-id` attribute.
 *
 * Now that the Masonry Gallery Block contains inner Image Blocks,
 * we add a custom `data-id` attribute before rendering the masonry gallery
 * so that the Image Block can pick it up in its render_callback.
 *
 * @param  array $parsed_block A single parsed block object.
 *
 * @return array               The migrated block object.
 */
function render_block_coblocks_masonry_gallery_data( $parsed_block ) {
	if ( 'coblocks/gallery-masonry' === $parsed_block['blockName'] ) {
		foreach ( $parsed_block['innerBlocks'] as $key => $inner_block ) {
			if ( 'core/image' === $inner_block['blockName'] ) {
				if ( ! isset( $parsed_block['innerBlocks'][ $key ]['attrs']['data-id'] ) && isset( $inner_block['attrs']['id'] ) ) {
					$parsed_block['innerBlocks'][ $key ]['attrs']['data-id'] = esc_attr( $inner_block['attrs']['id'] );
				}
			}
		}
	}

	return $parsed_block;
}

add_filter( 'render_block_data', 'render_block_coblocks_masonry_gallery_data' );

/**
 * Registers the `coblocks/gallery-masonry` block on server.
 * This render callback needs to be here
 * so that the gallery styles are loaded in block-based themes.
 */
function gutenberg_register_block_coblocks_masonry_gallery() {
	register_block_type_from_metadata(
		__DIR__ . '/gallery-masonry',
		array(
			'render_callback' => function ( $attributes, $content ) {
				return $content;
			},
		)
	);
}

add_action( 'init', 'gutenberg_register_block_coblocks_masonry_gallery', 20 );
