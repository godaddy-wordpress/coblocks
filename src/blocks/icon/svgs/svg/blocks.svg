<svg fill="none" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="m10.3519669 11.7141378v8.2858622h8.2815735v-8.2858622zm-10.3519669 8.2858622h8.2815735v-8.2858622h-8.2815735zm0-18.64319006v8.28586225h8.2815735v-8.28586225zm14.1407867-1.35680994-5.8592132 5.85189021 5.8592132 5.86224759 5.8592133-5.86224759z" fill="currentColor" /></svg>