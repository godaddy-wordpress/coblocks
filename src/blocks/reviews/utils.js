/**
 * External dependencies
 */
import { StarStyles } from '@godaddy-wordpress/coblocks-icons';

/**
 * WordPress dependencies
 */
import { Icon } from '@wordpress/components';

export const renderRating = ( rating, baseRating = 5 ) => {
	const ui = [];

	ui.push( ...Array( rating ).fill( renderRatingIcon( StarStyles.default ) ) );

	if ( baseRating - rating > 0 ) {
		ui.push( ...Array( baseRating - rating ).fill( renderRatingIcon( StarStyles.outlined ) ) );
	}

	return ui;
};

const renderRatingIcon = ( ratingIcon ) => <Icon icon={ ratingIcon } size={ 16 } />;
