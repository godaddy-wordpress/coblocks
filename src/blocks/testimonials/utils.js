/**
 * External dependencies.
 */
import { find } from 'lodash';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import TokenList from '@wordpress/token-list';

export const blockStyles = [
	{
		isDefault: true,
		/* translators: block style */
		label: __( 'Boxy', 'coblocks' ),
		name: 'tall',
	},
	{
		/* translators: block style */
		label: __( 'Conversation', 'coblocks' ),
		name: 'conversation',
	},
	{
		/* translators: block style */
		label: __( 'Horizontal', 'coblocks' ),
		name: 'horizontal',
	},
];

/**
 * Returns the active style from the given className.
 *
 * @param {Array}  styles    Block style variations.
 * @param {string} className Class name
 * @return {Object?} The active style.name.
 */
export const getActiveStyle = ( styles, className ) => {
	for ( const style of new TokenList( className ).values() ) {
		if ( style.indexOf( 'is-style-' ) === -1 ) {
			continue;
		}

		const potentialStyleName = style.substring( 9 );
		const activeStyle = find( styles, { name: potentialStyleName } );

		if ( activeStyle ) {
			return activeStyle.name;
		}
	}

	return find( styles, 'isDefault' ).name;
};
