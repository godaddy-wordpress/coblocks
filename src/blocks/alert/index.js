/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const { replaceBlocks } = dispatch( 'core/block-editor' );

		replaceBlocks(
			[ props.clientId ],
			createBlock( 'core/paragraph', props.attributes ),
		);
		return null;
	},
	parent: [],
	save: () => null,
	title: metadata.title,
};

export { name, metadata, settings };
