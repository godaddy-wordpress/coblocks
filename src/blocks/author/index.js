/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch, select } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const parentBlock = select( 'core/editor' ).getBlocksByClientId( props.clientId )[ 0 ];
		if ( ! parentBlock ) {
			return null;
		}

		// Used to set text color props
		const hasTextColor = ( props.attributes.hasOwnProperty( 'textColor' ) );
		const hasCustomTextColor = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'color' ) && props.attributes.style.color.hasOwnProperty( 'text' ) );

		// Used to set background color props
		const hasBackgroundColor = ( props.attributes.hasOwnProperty( 'backgroundColor' ) );
		const hasCustomBackgroundColor = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'color' ) && props.attributes.style.color.hasOwnProperty( 'background' ) );

		// Used to set font props
		const hasFontSize = ( props.attributes.hasOwnProperty( 'fontSize' ) );
		const hasCustomFont = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'typography' ) );

		const { replaceBlocks } = dispatch( 'core/block-editor' );
		const isRTL = select( 'core/editor' ).getEditorSettings().isRTL;

		const authorNameBlock = () => {
			let authorNameBlockProps = {
				content: props.attributes.name,
				level: 4,
			};

			// Font controls - default is fontSize and style is used for custom setting.
			// All font controls should apply to the bio and name.
			if ( hasFontSize ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					fontSize: props.attributes.fontSize,
				};
			}
			if ( hasCustomFont ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					style: { ...authorNameBlockProps?.style,
						typography: props.attributes.style.typography,
					},
				};
			}

			// Headings do not inherit color from parent - we descend the style here for initial match.
			if ( hasCustomTextColor ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					style: { ...authorNameBlockProps?.style,
						color: { ...authorNameBlockProps?.style?.color,
							text: props.attributes.style.color.text,
						},
					},
				};
			}
			if ( hasTextColor ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					textColor: props.attributes.textColor,
				};
			}

			return createBlock( 'core/paragraph', authorNameBlockProps );
		};

		const authorBioBlock = () => {
			let authorBioBlockProps = {
				content: props.attributes.biography,
			};

			// Font controls - default is fontSize and style is used for custom setting.
			// All font controls should apply to the bio and name.
			if ( hasFontSize ) {
				authorBioBlockProps = {
					...authorBioBlockProps,
					fontSize: props.attributes.fontSize,
				};
			}
			if ( hasCustomFont ) {
				authorBioBlockProps = {
					...authorBioBlockProps,
					style: {
						...authorBioBlockProps?.style,
						typography: props.attributes.style.typography,
					},
				};
			}

			return createBlock( 'core/paragraph', authorBioBlockProps );
		};

		const buttonBlock = parentBlock.innerBlocks;
		// Set buttons to align.
		buttonBlock.forEach( function( button, index ) {
			buttonBlock[ index ].attributes.align = isRTL ? 'right' : 'left';
		} );
		// Button blocks are no-longer stand-alone and should be within `core/buttons` block.
		const coreButtons = createBlock( 'core/buttons', { }, [ ...buttonBlock ] );

		const imageBlock = createBlock( 'core/image', { className: 'is-style-rounded', id: props.attributes.imgId, url: props.attributes.imgUrl } );

		const leftColumn = createBlock( 'core/column', { width: '25%' }, [ imageBlock ] );
		const rightColumn = createBlock( 'core/column', { width: '75%' }, [ authorNameBlock(), authorBioBlock(), coreButtons ] );

		const columnsBlock = () => {
			let columnsBlockProps = {
				...( props.attributes.hasOwnProperty( 'className' ) && { className: props.attributes.className } ),
				...( props.attributes.hasOwnProperty( 'animation' ) && { animation: props.attributes.animation } ),
				...( props.attributes.hasOwnProperty( 'anchor' ) && { anchor: props.attributes.anchor } ),
				style: {
					spacing: {
						padding: {
							bottom: '2.5rem',
							left: '2.5rem',
							right: '2.5rem',
							top: '2.5rem',
						},
					},
				},
			};

			// Migrate text colors into columns block props.
			if ( hasCustomTextColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							text: props.attributes.style.color.text,
						},
					},
				};
			}
			if ( hasTextColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					textColor: props.attributes.textColor,
				};
			}

			// Migrate background colors into columns block props.
			if ( hasCustomBackgroundColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							background: props.attributes.style.color.background,
						},
					},
				};
			}
			if ( hasBackgroundColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					backgroundColor: props.attributes.backgroundColor,
				};
			}

			// Does not have background color set. Use default block color from CoBlocks.
			const hasNoBackgroundColorSet = ! props.attributes.hasOwnProperty( 'backgroundColor' ) &&
				! props.attributes?.style?.color?.hasOwnProperty( 'background' );
			if ( hasNoBackgroundColorSet ) {
				// $dark-opacity-light-200 computed #8C8C971A - no variable access in root.
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							background: '#8C8C971A',
						},
					},
				};
			}

			return createBlock( 'core/columns', columnsBlockProps, [ leftColumn, rightColumn ] );
		};

		replaceBlocks(
			[ props.clientId ],
			columnsBlock()
		);

		return null;
	},
	parent: [],
	save: () => null,
	/* translators: block name */
	title: metadata.title,
};

export { name, metadata, settings };
