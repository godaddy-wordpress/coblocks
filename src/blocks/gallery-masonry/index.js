/**
 * External dependencies
 */
import { GalleryMasonryIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Icon, Spinner } from '@wordpress/components';
import { lazy, Suspense } from '@wordpress/element';

/**
 * Internal dependencies
 */
import deprecated from './deprecated';
const Edit = lazy( () => import( './edit' ) );
import { GalleryAttributes } from '../../components/block-gallery/shared';
import { hasFormattingCategory } from '../../utils/block-helpers';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * Block constants
 */
const { name, category } = metadata;

const attributes = {
	...GalleryAttributes,
	...metadata.attributes,
};

const settings = {
	attributes,
	category: hasFormattingCategory ? 'coblocks-galleries' : 'media',
	deprecated,
	/* translators: block description */
	description: __( 'Display multiple images in an organized masonry gallery.', 'coblocks' ),
	edit: ( props ) => (
		<Suspense fallback={ <Spinner /> }>
			<Edit { ...props } />
		</Suspense>
	),
	example: {
		innerBlocks: [
			{
				attributes: { url: 'https://s.w.org/images/core/5.3/Sediment_off_the_Yucatan_Peninsula.jpg' },
				name: 'core/image',
			},
			{
				attributes: { url: 'https://s.w.org/images/core/5.3/Windbuchencom.jpg' },
				name: 'core/image',

			},
			{
				attributes: { url: 'https://s.w.org/images/core/5.3/Biologia_Centrali-Americana_-_Cantorchilus_semibadius_1902.jpg' },
				name: 'core/image',

			},
			{
				attributes: { url: 'https://s.w.org/images/core/5.3/MtBlanc1.jpg' },
				name: 'core/image',

			},
			{
				attributes: { url: 'hhttps://s.w.org/images/core/5.3/Glacial_lakes,_Bhutan.jpg' },
				name: 'core/image',
			},
		],
	},
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'gallery', 'coblocks' ),
		/* translators: block keyword */
		__( 'photos', 'coblocks' ),
		/* translators: block keyword */
		__( 'lightbox', 'coblocks' ),
	],
	save,
	/* translators: block name */
	title: __( 'Masonry', 'coblocks' ),
	transforms,
};

export { name, category, metadata, settings };
