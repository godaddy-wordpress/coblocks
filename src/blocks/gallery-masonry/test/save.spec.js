/**
 * WordPress dependencies
 */
import { registerCoreBlocks } from '@wordpress/block-library';
import { createBlock, serialize } from '@wordpress/blocks';

registerCoreBlocks();

/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';

/**
 * Internal dependencies.
 */
import * as settings from '../index';
import { registerBlock } from '../../../../src/utils/helper';

// Make variables accessible for all tests.
let block;
let serializedBlock;

const baseAttributes = {
	gutter: 'medium',
};

const baseInnerBlocks = [
	createBlock( 'core/image', {
		url: 'https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg',
		id: 1,
		link: 'https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg',
	} ),
	createBlock( 'core/image', {
		url: 'https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg',
		id: 2,
		link: 'https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg',
	} ),
];

describe( 'coblocks/gallery-masonry', () => {
	beforeAll( () => {
		// Register the block.
		registerBlock( settings );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( settings.name, baseAttributes, baseInnerBlocks );
		// Reset the reused variables.
		serializedBlock = '';
	} );

	it( 'should have `core/image` innerBlocks rendered', () => {
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'src="https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg"' );
		expect( serializedBlock ).toContain( 'wp:image' );

		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should have className \'has-lightbox\' with lightbox enabled.', () => {
		block.attributes = { ...block.attributes, lightbox: true };
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'has-lightbox' );

		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should have caption with `core/image` and caption with `coblocks/masonry`.', () => {
		block.innerBlocks = [
			...block.innerBlocks,
			createBlock( 'core/image', {
				url: 'https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg',
				id: 3,
				link: 'https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg',
				caption: 'Test Caption',
			} ),
		];

		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( `src="https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg"` );
		expect( serializedBlock ).toContain( '<figcaption>Test Caption</figcaption>' );
		expect( serializedBlock ).not.toContain( '<figcaption class="blocks-gallery-caption">' );

		block.attributes.caption = 'Masonry Caption';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toContain( '<figcaption class="blocks-gallery-caption">' );
		expect( serializedBlock ).toContain( 'Masonry Caption' );

		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should have className \'has-border-radius-10\' with radius set to 10.', () => {
		block.attributes = { ...block.attributes, radius: 0 };
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).not.toContain( 'has-border-radius-10' );

		block.attributes = { ...block.attributes, radius: 10 };
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'has-border-radius-10' );

		expect( serializedBlock ).toMatchSnapshot();
	} );

	[ 'small', 'medium', 'large' ].forEach( ( gutter ) => {
		it( `should have className 'has-${ gutter }-gutter' with gutter set to '${ gutter }'.`, () => {
			block.attributes = { ...block.attributes, gutter };

			serializedBlock = serialize( block );

			expect( serializedBlock ).toBeDefined();
			expect( serializedBlock ).toContain( `has-${ gutter }-gutter` );

			expect( serializedBlock ).toMatchSnapshot();
		} );
	} );

	[ 'grayscale', 'sepia', 'saturation', 'dim', 'vintage' ].forEach( ( filter ) => {
		it( `should have className \'has-filter-${ filter }\' with filter set to '${ filter }'.`, () => {
			block.attributes = { ...block.attributes, filter };
			serializedBlock = serialize( block );

			expect( serializedBlock ).toBeDefined();
			expect( serializedBlock ).toContain( `has-filter-${ filter }` );

			expect( serializedBlock ).toMatchSnapshot();
		} );
	} );
} );
