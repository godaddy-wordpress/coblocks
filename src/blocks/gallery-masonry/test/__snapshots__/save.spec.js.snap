// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/gallery-masonry should have \`core/image\` innerBlocks rendered 1`] = `
"<!-- wp:coblocks/gallery-masonry {\\"gutter\\":\\"medium\\"} -->
<figure class=\\"wp-block-coblocks-gallery-masonry masonry-grid has-medium-gutter\\"><!-- wp:image {\\"id\\":1} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\" alt=\\"\\" class=\\"wp-image-1\\"/></figure>
<!-- /wp:image -->

<!-- wp:image {\\"id\\":2} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\" alt=\\"\\" class=\\"wp-image-2\\"/></figure>
<!-- /wp:image --></figure>
<!-- /wp:coblocks/gallery-masonry -->"
`;

exports[`coblocks/gallery-masonry should have caption with \`core/image\` and caption with \`coblocks/masonry\`. 1`] = `
"<!-- wp:coblocks/gallery-masonry {\\"gutter\\":\\"medium\\"} -->
<figure class=\\"wp-block-coblocks-gallery-masonry masonry-grid has-medium-gutter\\"><!-- wp:image {\\"id\\":1} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\" alt=\\"\\" class=\\"wp-image-1\\"/></figure>
<!-- /wp:image -->

<!-- wp:image {\\"id\\":2} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\" alt=\\"\\" class=\\"wp-image-2\\"/></figure>
<!-- /wp:image -->

<!-- wp:image {\\"id\\":3} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\" alt=\\"\\" class=\\"wp-image-3\\"/><figcaption>Test Caption</figcaption></figure>
<!-- /wp:image --></figure><figcaption class=\\"blocks-gallery-caption\\">Masonry Caption</figcaption>
<!-- /wp:coblocks/gallery-masonry -->"
`;

exports[`coblocks/gallery-masonry should have className 'has-border-radius-10' with radius set to 10. 1`] = `
"<!-- wp:coblocks/gallery-masonry {\\"gutter\\":\\"medium\\",\\"radius\\":10} -->
<figure class=\\"wp-block-coblocks-gallery-masonry masonry-grid has-border-radius-10 has-medium-gutter\\"><!-- wp:image {\\"id\\":1} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\" alt=\\"\\" class=\\"wp-image-1\\"/></figure>
<!-- /wp:image -->

<!-- wp:image {\\"id\\":2} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\" alt=\\"\\" class=\\"wp-image-2\\"/></figure>
<!-- /wp:image --></figure>
<!-- /wp:coblocks/gallery-masonry -->"
`;

exports[`coblocks/gallery-masonry should have className 'has-filter-dim' with filter set to 'dim'. 1`] = `
"<!-- wp:coblocks/gallery-masonry {\\"filter\\":\\"dim\\",\\"gutter\\":\\"medium\\"} -->
<figure class=\\"wp-block-coblocks-gallery-masonry masonry-grid has-filter-dim has-medium-gutter\\"><!-- wp:image {\\"id\\":1} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\" alt=\\"\\" class=\\"wp-image-1\\"/></figure>
<!-- /wp:image -->

<!-- wp:image {\\"id\\":2} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\" alt=\\"\\" class=\\"wp-image-2\\"/></figure>
<!-- /wp:image --></figure>
<!-- /wp:coblocks/gallery-masonry -->"
`;

exports[`coblocks/gallery-masonry should have className 'has-filter-grayscale' with filter set to 'grayscale'. 1`] = `
"<!-- wp:coblocks/gallery-masonry {\\"filter\\":\\"grayscale\\",\\"gutter\\":\\"medium\\"} -->
<figure class=\\"wp-block-coblocks-gallery-masonry masonry-grid has-filter-grayscale has-medium-gutter\\"><!-- wp:image {\\"id\\":1} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\" alt=\\"\\" class=\\"wp-image-1\\"/></figure>
<!-- /wp:image -->

<!-- wp:image {\\"id\\":2} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\" alt=\\"\\" class=\\"wp-image-2\\"/></figure>
<!-- /wp:image --></figure>
<!-- /wp:coblocks/gallery-masonry -->"
`;

exports[`coblocks/gallery-masonry should have className 'has-filter-saturation' with filter set to 'saturation'. 1`] = `
"<!-- wp:coblocks/gallery-masonry {\\"filter\\":\\"saturation\\",\\"gutter\\":\\"medium\\"} -->
<figure class=\\"wp-block-coblocks-gallery-masonry masonry-grid has-filter-saturation has-medium-gutter\\"><!-- wp:image {\\"id\\":1} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\" alt=\\"\\" class=\\"wp-image-1\\"/></figure>
<!-- /wp:image -->

<!-- wp:image {\\"id\\":2} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\" alt=\\"\\" class=\\"wp-image-2\\"/></figure>
<!-- /wp:image --></figure>
<!-- /wp:coblocks/gallery-masonry -->"
`;

exports[`coblocks/gallery-masonry should have className 'has-filter-sepia' with filter set to 'sepia'. 1`] = `
"<!-- wp:coblocks/gallery-masonry {\\"filter\\":\\"sepia\\",\\"gutter\\":\\"medium\\"} -->
<figure class=\\"wp-block-coblocks-gallery-masonry masonry-grid has-filter-sepia has-medium-gutter\\"><!-- wp:image {\\"id\\":1} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\" alt=\\"\\" class=\\"wp-image-1\\"/></figure>
<!-- /wp:image -->

<!-- wp:image {\\"id\\":2} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\" alt=\\"\\" class=\\"wp-image-2\\"/></figure>
<!-- /wp:image --></figure>
<!-- /wp:coblocks/gallery-masonry -->"
`;

exports[`coblocks/gallery-masonry should have className 'has-filter-vintage' with filter set to 'vintage'. 1`] = `
"<!-- wp:coblocks/gallery-masonry {\\"filter\\":\\"vintage\\",\\"gutter\\":\\"medium\\"} -->
<figure class=\\"wp-block-coblocks-gallery-masonry masonry-grid has-filter-vintage has-medium-gutter\\"><!-- wp:image {\\"id\\":1} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\" alt=\\"\\" class=\\"wp-image-1\\"/></figure>
<!-- /wp:image -->

<!-- wp:image {\\"id\\":2} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\" alt=\\"\\" class=\\"wp-image-2\\"/></figure>
<!-- /wp:image --></figure>
<!-- /wp:coblocks/gallery-masonry -->"
`;

exports[`coblocks/gallery-masonry should have className 'has-large-gutter' with gutter set to 'large'. 1`] = `
"<!-- wp:coblocks/gallery-masonry {\\"gutter\\":\\"large\\"} -->
<figure class=\\"wp-block-coblocks-gallery-masonry masonry-grid has-large-gutter\\"><!-- wp:image {\\"id\\":1} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\" alt=\\"\\" class=\\"wp-image-1\\"/></figure>
<!-- /wp:image -->

<!-- wp:image {\\"id\\":2} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\" alt=\\"\\" class=\\"wp-image-2\\"/></figure>
<!-- /wp:image --></figure>
<!-- /wp:coblocks/gallery-masonry -->"
`;

exports[`coblocks/gallery-masonry should have className 'has-lightbox' with lightbox enabled. 1`] = `
"<!-- wp:coblocks/gallery-masonry {\\"gutter\\":\\"medium\\",\\"lightbox\\":true} -->
<figure class=\\"wp-block-coblocks-gallery-masonry masonry-grid has-lightbox has-medium-gutter\\"><!-- wp:image {\\"id\\":1} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\" alt=\\"\\" class=\\"wp-image-1\\"/></figure>
<!-- /wp:image -->

<!-- wp:image {\\"id\\":2} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\" alt=\\"\\" class=\\"wp-image-2\\"/></figure>
<!-- /wp:image --></figure>
<!-- /wp:coblocks/gallery-masonry -->"
`;

exports[`coblocks/gallery-masonry should have className 'has-medium-gutter' with gutter set to 'medium'. 1`] = `
"<!-- wp:coblocks/gallery-masonry {\\"gutter\\":\\"medium\\"} -->
<figure class=\\"wp-block-coblocks-gallery-masonry masonry-grid has-medium-gutter\\"><!-- wp:image {\\"id\\":1} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\" alt=\\"\\" class=\\"wp-image-1\\"/></figure>
<!-- /wp:image -->

<!-- wp:image {\\"id\\":2} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\" alt=\\"\\" class=\\"wp-image-2\\"/></figure>
<!-- /wp:image --></figure>
<!-- /wp:coblocks/gallery-masonry -->"
`;

exports[`coblocks/gallery-masonry should have className 'has-small-gutter' with gutter set to 'small'. 1`] = `
"<!-- wp:coblocks/gallery-masonry {\\"gutter\\":\\"small\\"} -->
<figure class=\\"wp-block-coblocks-gallery-masonry masonry-grid has-small-gutter\\"><!-- wp:image {\\"id\\":1} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\" alt=\\"\\" class=\\"wp-image-1\\"/></figure>
<!-- /wp:image -->

<!-- wp:image {\\"id\\":2} -->
<figure class=\\"wp-block-image\\"><img src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\" alt=\\"\\" class=\\"wp-image-2\\"/></figure>
<!-- /wp:image --></figure>
<!-- /wp:coblocks/gallery-masonry -->"
`;
